src/osd/PrimaryLogPG.cc

```diff
@@ -3472,6 +3474,29 @@ void PrimaryLogPG::dec_refcount(ObjectContextRef obc, const object_ref_delta_t&
   }
 }
 
+void PrimaryLogPG::dec_refcount_by_dirty(OpContext* ctx, const object_manifest_t& updated_chunks)
+{
+  object_ref_delta_t refs;
+  ObjectContextRef cobc = nullptr;
+  ObjectContextRef obc = ctx->obc;
+  // Look over previous snapshot, then figure out whether updated chunk needs to be deleted
+  if (obc->ssc->snapset.clones.size()) {
```

Comment:
Your get_prev_clone_obc already handles the case where there isn't a previous snapshot, why not  include this check there?